libcontainer/container_linux.go

```diff
@@ -1173,16 +1174,21 @@ func (c *linuxContainer) restoreNetwork(req *criurpc.CriuReq, criuOpts *CriuOpts
 // rootfs_linux.go
 func (c *linuxContainer) makeCriuRestoreMountpoints(m *configs.Mount) error {
 	switch m.Device {
-	case "cgroup":
-		// Do nothing for cgroup, CRIU should handle it
 	case "bind":
 		// The prepareBindMount() function checks if source
 		// exists. So it cannot be used for other filesystem types.
 		if err := prepareBindMount(m, c.config.Rootfs); err != nil {
 			return err
 		}
+	case "cgroup":
+		if !cgroups.IsCgroup2UnifiedMode() {
+			// mount points are saved by CRIU as /sys/fs/cgroup is tmpfs
+			return nil
+		}
+		// /sys/fs/cgroup is a real mount, the mountpoint need to be created
+		fallthrough
```

Comment:
So I tried all (a lot) combinations. In no case is it necessary to create the base directory `/sys/fs/cgroup` as that exists as soon as sysfs is mounted. For V1 we always have a tmpfs on `/sys/fs/cgroup` which means that we do not need to create any directories for CRIU. I think the fallthrough is not necessary. But is also does not hurt.

Reply:
Interesting, maybe I was wrong here. And you're right, even if we create /sys/fs/cgroup, once /sys is mounted, our dir is shadowed. Let's retry without fallthrough.

The commit (including the commit message) changed accordingly.

Reply:
Yep, I was obviously wrong, thanks, @adrianreber!